server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/DatafileManager.java

```diff
@@ -417,6 +425,8 @@ Optional<StoredTabletFile> bringMinorCompactionOnline(TabletFile tmpDatafile,
           }
           datafileSizes.put(newFileStored, dfv);
         }
+        tabletLog.flushed(newFile);
+        checkTransactionLog();
```

Comment:
I wonder if `checkTransactionLog` should be called inside of `Tablet.compareTabletInfo` which is called from the tablet health checker thread. This would allow you to remove the size property, just keep all of the transactions between the calls. When `checkTransactionLog` is called, then clear the log and dump the contents if they are not what is expected. If you do this, then I would replace the property that exists now, which a boolean property that enables/disables this feature.

Reply:
The log is simply so that we can more readily figure out what has happened when something goes wrong.  Whenever we detect that something is wrong, I want to see the last N transactions.  I do not want to clear the transactions but keep them around.

Reply:
Setting the maxSize for the log to 0 effectively disables any useful information from the log.  I suppose I could disable any logging in that case.

Reply:
ok,, logging of the transactions are disabled if there are no transactions.  So setting transaction log size of 0 effectively disables it.